rotkehlchen/chain/ethereum/adex/typing.py

```diff
@@ -0,0 +1,72 @@
+from enum import Enum
+from typing import Any, Dict, NamedTuple
+
+from eth_typing.evm import ChecksumAddress
+from eth_utils.typing import HexStr
+
+from rotkehlchen.accounting.structures import Balance
+from rotkehlchen.assets.asset import EthereumToken
+from rotkehlchen.fval import FVal
+from rotkehlchen.typing import Timestamp
+
+
+class EventType(Enum):
+    """Supported events"""
+    BOND = 1
+    UNBOND = 2
+    UNBOND_REQUEST = 3
+
+    def __str__(self) -> str:
+        if self == EventType.BOND:
+            return 'bond'
+        if self == EventType.UNBOND:
+            return 'unbond'
+        if self == EventType.UNBOND_REQUEST:
+            return 'unbond_request'
+        raise RuntimeError(f'Corrupt value {self} for EventType -- Should never happen')
+
+
+class Bond(NamedTuple):
+    tx_hash: HexStr  # from bond.id
+    address: ChecksumAddress
+    identity_address: ChecksumAddress
+    bond_id: HexStr
+    amount: FVal
+    pool_id: HexStr
+    slashed_at_start: Timestamp  # FVal
+    timestamp: Timestamp
+    event_type: EventType = EventType.BOND
+
+
+class Unbond(NamedTuple):
+    tx_hash: HexStr  # from unbond.id
+    address: ChecksumAddress
+    identity_address: ChecksumAddress
+    bond_id: HexStr
+    timestamp: Timestamp
+    event_type: EventType = EventType.UNBOND
+
+
+class UnbondRequest(NamedTuple):
+    tx_hash: HexStr  # from unbond.id
+    address: ChecksumAddress
+    identity_address: ChecksumAddress
+    bond_id: HexStr
+    will_unlock: FVal
+    timestamp: Timestamp
+    event_type: EventType = EventType.UNBOND_REQUEST
+
+
+class PoolBalance(NamedTuple):
```

Comment:
It's not really a pool as you are not staking multiple tokens. It's just a staking contract and you are always staking ADX. So I think this should be simplified to `ADXStakingBalance` and remove the token as it's always ADX.

Unless you want a generic `StakingBalance` data structure ... but in that case pool_id being a HexStr and pool_name are already very specific